modules/service/src/main/scala/lucuma/odb/service/ObservationWorkflowService.scala

```diff
@@ -0,0 +1,461 @@
+            OptionT.fromOption(Option.unless(ra.isInInterval(raStart, raEnd) && dec.isInInterval(decStart, decEnd))(ObservationValidation.callForProposals(Messages.ExplicitBaseOutOfRange)))
+          }
+        } yield validation).value
+
+      /* Validate that an observation is compatible with its program's CFP. */
+      private def cfpValidations(info: ObservationValidationInfo): F[ObservationValidationMap] =
+        info.cfpid.fold(ObservationValidationMap.empty.pure): cid =>
+          for
+            valInstr        <- validateInstrument(cid, info.instrument)
+            explicitBase     = (info.ra, info.dec).tupled
+            valRaDec        <- validateRaDec(info, cid, explicitBase)
+            valForactivation = Option.when(info.forReview)(ObservationValidation.configuration(Messages.ConfigurationForReview))
+          yield ObservationValidationMap.fromList(List(valInstr, valRaDec, valForactivation).flatten)
+
+      private def validateInstrument(cid: CallForProposals.Id, optInstr: Option[Instrument]): F[Option[ObservationValidation]] = {
+        // If there is no instrument in the observation, that will get caught with the generatorValidations
+        optInstr.fold(none.pure){ instr =>
+          session.stream(Statements.CfpInstruments)(cid, chunkSize = 1024)
+            .compile
+            .toList
+            .map(l =>
+              if(l.isEmpty || l.contains(instr)) none
+              else ObservationValidation.callForProposals(Messages.invalidInstrument(instr)).some
+            )
+        }
+      }
+
+      private def obsInfo(oid: Observation.Id): F[ObservationValidationInfo] =
+        session.unique(Statements.ObservationValidationInfo)(oid)
+
+      private def itcValidations(
+        info: ObservationValidationInfo,
+        itcClient: ItcClient[F], 
+        params: GeneratorParams
+      )(using Transaction[F]): F[ObservationValidationMap] =
+        itcService(itcClient).selectOne(info.pid, info.oid, params).map:
+          // N.B. there will soon be more cases here
+          case Some(_) => ObservationValidationMap.empty
+          case None    => ObservationValidationMap.singleton(ObservationValidation.itc("ITC results are not present."))
+
+      private def validateScienceBand(oid: Observation.Id): F[ObservationValidationMap] =
+        session
+          .option(Statements.SelectInvalidBand)(oid)
+          .map { invalidBand =>
+            val m = ObservationValidationMap.empty
+            invalidBand.fold(m)(b => m.add(ObservationValidation.configuration(Messages.invalidScienceBand(b))))
+          }
+
+      private def validateConfiguration(oid: Observation.Id)(using Transaction[F]): F[ObservationValidationMap] =
+        configurationService.selectRequests(oid).map: r =>
+          val m = ObservationValidationMap.empty
+          r.toOption match
+            case None => m.add(ObservationValidation.configuration(Messages.ConfigurationRequest.Unavailable))
```

Comment:
It seems this isn't using the new ObservationValidation instances, yet?

Reply:
Oh, huh. You are right. I thought I updated core. Standby.

Reply:
Also, I think the ObservationValidation enum needs to be updated in the Schema?

Reply:
Ok updating that reveals that one of the tags is malformed in `ObservationValidationCode` in core. Need to update again.

Reply:
https://github.com/gemini-hlsw/lucuma-core/pull/963